src/auth/auth.controller.ts

```diff
@@ -30,72 +22,150 @@ import {
   ApiOperation,
   ApiForbiddenResponse,
 } from '@nestjs/swagger';
-import { LocalAuthGuard } from './guards';
+import { RabbitRPC } from '@golevelup/nestjs-rabbitmq';
+import { ReplyErrorCallback } from './error-callback';
 
 @Controller('auth')
 @ApiTags('Authentication')
 export class AuthController {
   constructor(private readonly authService: AuthService) {}
 
+  @RabbitRPC({
+    exchange: 'authentication',
+    routingKey: 'signup',
+    errorHandler: ReplyErrorCallback,
+  })
   @HttpCode(201)
   @ApiOperation({ summary: 'Add user to database' })
   @ApiBadRequestResponse({ description: 'Wrong credentials provided' })
   @ApiForbiddenResponse({
     description: 'Cannot add User to database, use different credentials',
   })
-  @Post('signup')
-  async signUp(@Body() signUpRequest: SignUpRequest): Promise<SignUpResponse> {
-    const userId = await this.authService.signUp(signUpRequest);
-    return { userId };
+  @RabbitRPC({
+    exchange: 'authentication',
+    routingKey: 'signup',
+    errorHandler: ReplyErrorCallback,
+  })
+  async signUp(test: SignUpRequest): Promise<SignUpResponse> {
+    const userId = await this.authService.signUp(test);
+
+    const accountActivationToken =
+      await this.authService.generateAccountActivationToken(userId);
+
+    return SignUpResponse.from(accountActivationToken);
   }
+
   @HttpCode(200)
-  @UseGuards(LocalAuthGuard)
   @ApiOperation({ summary: 'Authenticate user' })
-  @Post('signin')
-  async signIn(@Body() signInRequest: SignInRequest): Promise<SignInResponse> {
+  @RabbitRPC({
+    exchange: 'authentication',
+    routingKey: 'signin',
+    errorHandler: ReplyErrorCallback,
+  })
+  async signIn(signInRequest: SignInRequest): Promise<SignInResponse> {
     const accessToken = await this.authService.signIn(signInRequest);
 
     return SignInResponse.from(accessToken);
   }
 
+  @HttpCode(200)
+  @ApiOperation({ summary: 'Validate user' })
+  @RabbitRPC({
+    exchange: 'authentication',
+    routingKey: 'validate-user',
+    errorHandler: ReplyErrorCallback,
+  })
+  async validateUser(validationData: SignInRequest): Promise<number> {
+    const validatedUserId = await this.authService.validateUser(validationData);
+
+    return validatedUserId;
+  }
+
+  @HttpCode(200)
+  @ApiOperation({ summary: 'Validate jwt Token' })
```

Comment:
In descriptions etc. use JWT, as it's abbreviation from JSON Web token so:
```suggestion
  @ApiOperation({ summary: 'Validate JWT Token' })
```